Add ability to present circular menu buttons in a counterclockwise layout

jkanalakis opened the pull request:
There is a use case in which presenting circular menu buttons in a counterclockwise fashion is very helpful, especially when the FloatingButton is located at the top of a view instead of the bottom of a view. This change optionally reverses the path and demonstrates the change in the FloatingButtonExample code.

jkanalakis committed:
Add ability to present circular menu buttons in a counterclockwise layout

There is a use case in which presenting circular menu buttons in a counterclockwise fashion is very helpful, especially when the FloatingButton is located at the top of a view instead of the bottom of a view. This change reverses the path and demonstrates the change in the FloatingButtonExample code.

f3dm76 commented on Sources/FloatingButton/FloatingButton.swift:
```diff
@@ -12,6 +12,10 @@ public enum Direction {
 }
 
+public enum Rotation {
+    case clockwise, counterclockwise
```
could you please use camel case - counterClockwise

jkanalakis replied:
Excellent catch. ;)

jkanalakis replied:
BTW, I *really* like this project, it is extremely well done and I find it very useful!

f3dm76 commented on README.md:
```diff
@@ -70,6 +71,7 @@ FloatingButton(mainButtonView: mainButton, buttons: buttons, isOpen: $isOpen)
 `startAngle`  
 `endAngle`  
 `radius` - distance between center of main button and centers of submenu buttons  
+`rotation` - changes the button layout direction from the startAngle to the endAngle
```
I think 'rotation' is reserved for another use case. Since it's a button layout direction - would you please call it 'layoutDirection'?

jkanalakis replied:
Good idea!

f3dm76 reviewed (changes requested):
Great work, thank you! I just left a couple of small comments and we are good to go

jkanalakis committed:
Minor naming adjustments to comply with naming conventions

Updated counterClockwise name to adopt camel case and changed rotation to layoutDirection.

jkanalakis committed:
Correction to the markdown instructions

f3dm76 commented on Sources/FloatingButton/FloatingButton.swift:
```diff
@@ -12,6 +12,10 @@ public enum Direction {
     case left, right, top, bottom
 }
 
+public enum Rotation {
```
LayoutDirection

f3dm76 commented on Sources/FloatingButton/FloatingButton.swift:
```diff
@@ -44,6 +48,7 @@ public struct FloatingButton<MainView, ButtonView>: View where MainView: View, B
     fileprivate var alignment: Alignment = .center
     
     // circle
+    fileprivate var rotation: Rotation = .clockwise
```
layoutDirection

f3dm76 commented on Sources/FloatingButton/FloatingButton.swift:
```diff
@@ -391,6 +398,13 @@ public extension FloatingButtonGeneric where T: CircleFloatingButton {
         return copy
     }
+
+    func layoutDirection(_ rotation: Rotation) -> FloatingButtonGeneric {
```
layoutDirection: LayoutDirection

jkanalakis committed:
Additional renaming to layoutDirection that was missed

f3dm76 commented:
You changes are part of 1.3.0 release, have a wonderful day!

jkanalakis commented:
Thank you, I'm excited to contribute. I really like what the team at Exyte is doing!
